### PR TITLE
Generate next cycle test applications

### PR DESCRIPTION
**Pull request by JR-G**
## Context

We're launching the new references flow in the next cycle. This feature will only be shown to candidates who have a `recruitment_cycle_year` value of next cycle. This requires the user to manually update the recruitment cycle value on the application form. This task will generate next cycle applications in various states.

## Changes proposed in this pull request

New task in the support UI:
image

## Guidance to review

Maybe we don't care about the course itself for the purpose of out testing but i'm not a fan of:

```
courses = if year == RecruitmentCycle.next_year
                Course.open_on_apply.in_cycle(RecruitmentCycle.current_year)
              else
                Course.open_on_apply.in_cycle(year)
              end
```

Also need to consider the `POST /test-data/generate` endpoint, but perhaps for a follow up PR.

### CODE REVIEW

**Comment by JR-G** [comment]
[Generate next cycle applications task](https://trello.com/c/yHugeYaL/681-generate-next-cycle-applications-task)

File: app/views/support_interface/tasks/index.html.erb [hunk 1]
```diff
@@ -65,6 +65,18 @@
...
   </section>
 
+    <section class="app-section app-section--with-top-border">
+    <div class="govuk-grid-row">
+      <div class="govuk-grid-column-two-thirds">
+        <h2 class="govuk-heading-m">Generate next cycle test applications</h2>
+        <p class="govuk-body">This task generates mostly-random test applications for the next cycle.</p>
+      </div>
+      <div class="govuk-grid-column-one-third">
+        <%= govuk_button_to 'Generate next cycle test applications', support_interface_run_task_path('generate_next_cycle_test_applications'), secondary: true %>
```
Comment (frankieroberto) [inline app/views/support_interface/tasks/index.html.erb:75]: Can we make this more explicit which cycle it's referring to, as "next cycle" can easily get confusing, especially when using the switcher?

Suggest we add a reference to the year (eg "2022-2023") and explicitly say "recruitment cycle", as "cycle" also sometimes gets used to describe the academic years of the courses being applied to.

```suggestion
        <h2 class="govuk-heading-m">Generate test applications for the <%= cycle %> recruitment cycle</h2>
        <p class="govuk-body">This task generates 100 test applications for the <%= cycle %> recruitment cycle.</p>
      </div>
      <div class="govuk-grid-column-one-third">
        <%= govuk_button_to "Generate test applications for the #{cycle} recruitment cycle", support_interface_run_task_path('generate_next_cycle_test_applications'), secondary: true %>
```

**Review by frankieroberto — COMMENTED** [review]
Nice!

We could also review the generated data for test applications for the next cycle, as the references data should be different (ie not include the actual reference, and have 2+ references).

File: app/views/support_interface/tasks/index.html.erb (hunk 1 above)
Comment (JR-G) [reply app/views/support_interface/tasks/index.html.erb:75]: 👍

**Pull request merged**